### PR TITLE
tests: coverage improvements

### PR DESCRIPTION
**Pull request by ramakrishnapallala**
tests: crypto: rand32: Add test for z_early_boot_rand32_get
tests: kernel: device: Set device power state

### CODE REVIEW

File: tests/kernel/device/src/main.c [hunk 1]
```diff
@@ -78,6 +79,7 @@ void test_dummy_device_pm(void)
 	zassert_false((busy == 0), NULL);
 
 	device_busy_clear(dev);
+	device_set_power_state(dev, DEVICE_PM_ACTIVE_STATE);
```
Comment (nniranjhana) [inline tests/kernel/device/src/main.c]: Why do we set to active state here? If it is for coverage improvement, do we not need to test `device_get_power_state` too, after this? I can see `set` is already being used in tests/power, but not `get`.

**Review by andyross — COMMENTED** [review]
Confused about the removed test case.  Either it's a mistake or needs explanation.

File: tests/kernel/device/src/main.c [hunk 2]
```diff
@@ -69,14 +68,13 @@ void test_dummy_device_pm(void)
...
+	/* Set device state to DEVICE_PM_ACTIVE_STATE */
+	device_set_power_state(dev, DEVICE_PM_ACTIVE_STATE);
 	device_busy_set(dev);
 
 	busy = device_any_busy_check();
 	zassert_false((busy == 0), NULL);
 
-	busy = device_busy_check(dev);
-	zassert_false((busy == 0), NULL);
-
```
[the remark this answers is not in the document]
Comment (nashif) [reply tests/kernel/device/src/main.c]: yep, confusing, in your commit you say: "removed redundant check device_any_busy_check test." but I do not see this being removed.

**Review by nashif — CHANGES_REQUESTED** [review]

File: tests/kernel/device/src/main.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (ramakrishnapallala) [reply tests/kernel/device/src/main.c]: sorry it was my mistake, i didn't read the lines properly. will correct it.

File: tests/crypto/rand32/src/main.c [hunk 3]
```diff
@@ -17,6 +18,8 @@
 
 #define N_VALUES 10
 
+extern u32_t z_early_boot_rand32_get(void);
```
[the remark this answers is not in the document]
Comment (ramakrishnapallala) [reply tests/crypto/rand32/src/main.c]: as it was internal to kernel I was a bit hesitant to add the header. I will add it now.

File: tests/kernel/device/src/main.c (hunk 1 above)
Comment (ramakrishnapallala) [reply tests/kernel/device/src/main.c]: technically yes if the driver is implementing any suspend resume calls. But this is dummy driver which does not have those calls implemented. My intention of this change is to just invoke the device_pm_control_nop() through this call.

**Comment by ramakrishnapallala** [comment]
@nashif shippable is failing on mimxrt1050_evk of following error which is not related to the test:

/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/ext/hal/nxp/mcux/drivers/imx/./fsl_lpuart.h: In function ‘LPUART_WriteByte’:
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/ext/hal/nxp/mcux/drivers/imx/./fsl_lpuart.h:602:9: error: ‘LPUART_Type {aka struct <anonymous>}’ has no member named ‘data’
     base->DATA = data;
         ^~

**Review by nashif — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**